include/CppUTestExt/MockNamedValue.h

```diff
@@ -82,6 +82,7 @@ class MockNamedValue
     virtual void setValue(void* value);
     virtual void setValue(const void* value);
     virtual void setValue(const char* value);
+    virtual void setValue(const unsigned char* value);
```

Comment:
I initially tried to just use void* in the method signatures, and in MockNamedValue I added a method setValue(void* bufferPtr, size_t bufferSize), but then I found that internally the actual type is stored as a string instead of an enum or whatever (I presume that for an easy implementation of object types). 

So I'd had to come with a strange type name to set `type_`, like "memory buffer", "void* with size", "void* buffer" or whatever, and that didn't appeal me a lot. Maybe a bit influenced by my own programming guidelines (`char*` is for strings, `unsigned char*` for buffers which are not strings or which are generic), and considering that the "simple" type `unsigned char*` was not being overloaded, I decided to use it.

Anyway, taking in account that `MockNamedValue` is not used by framework users and that in `MockExpectedCall`and `MockActualCall` the actual `withParameter` method can't be accidentally miscalled because of the different signatures (i.e. the one with `unsigned char*` always takes an additional size parameter), I don't think that this is a problem. Anyway, the `setValue(const unsigned char* value)` could be replaced with a `setValue(const unsigned char* value, size_t size)` to avoid confusions.

Otherwise, it could be feasible to replace the actual `type_` with an enum and add an additional `objectType_` just for "user defined" types (i.e. for `withParameterOfType`).